certdepot/depot_test.go

```diff
@@ -0,0 +1,473 @@
+						assert.Equal(t, string(keyData), u.PrivateKey)
+						assert.Equal(t, string(certReqData), u.CertReq)
+						assert.Equal(t, user.CertRevocList, u.CertRevocList)
+						assert.Equal(t, newTTL, u.TTL)
+
+						certRevocListData := []byte("bob's new fake certificate revocation list")
+						assert.NoError(t, d.Put(depot.CrlTag(name), certRevocListData))
+						u = &User{}
+						assert.NoError(t, session.DB(databaseName).C(collectionName).FindId(name).One(u))
+						assert.Equal(t, name, u.ID)
+						assert.Equal(t, string(certData), u.Cert)
+						assert.Equal(t, string(keyData), u.PrivateKey)
+						assert.Equal(t, string(certReqData), u.CertReq)
+						assert.Equal(t, string(certRevocListData), u.CertRevocList)
+						assert.Equal(t, newTTL, u.TTL)
+					},
+				},
+				{
+					name: "CheckOnExistingUserWithNoData",
+					test: func(t *testing.T, d depot.Depot) {
+						name := "alice"
+						u := &User{
+							ID: name,
+						}
+						require.NoError(t, session.DB(databaseName).C(collectionName).Insert(u))
+
+						assert.False(t, d.Check(depot.CrtTag(name)))
+						assert.False(t, d.Check(depot.PrivKeyTag(name)))
+						assert.False(t, d.Check(depot.CsrTag(name)))
+						assert.False(t, d.Check(depot.CrlTag(name)))
+					},
+				},
+				{
+					name: "GetOnExistingUserWithNoData",
+					test: func(t *testing.T, d depot.Depot) {
+						name := "bob"
+						u := &User{
+							ID:  name,
+							TTL: time.Now(),
+						}
+						require.NoError(t, session.DB(databaseName).C(collectionName).Insert(u))
+
+						data, err := d.Get(depot.CrtTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+						data, err = d.Get(depot.PrivKeyTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+						data, err = d.Get(depot.CsrTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+						data, err = d.Get(depot.CrlTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+					},
+				},
+				{
+					name: "GetOnExpiredTTL",
+					test: func(t *testing.T, d depot.Depot) {
+						name := "bob"
+						u := &User{
+							ID:            name,
+							Cert:          "cert",
+							PrivateKey:    "key",
+							CertReq:       "certReq",
+							CertRevocList: "certRevocList",
+							TTL:           time.Time{},
+						}
+						require.NoError(t, session.DB(databaseName).C(collectionName).Insert(u))
+
+						data, err := d.Get(depot.CrtTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+						data, err = d.Get(depot.PrivKeyTag(name))
+						assert.NoError(t, err)
+						assert.Equal(t, u.PrivateKey, string(data))
+						data, err = d.Get(depot.CsrTag(name))
+						assert.NoError(t, err)
+						assert.Equal(t, u.CertReq, string(data))
+						data, err = d.Get(depot.CrlTag(name))
+						assert.Error(t, err)
+						assert.Nil(t, data)
+					},
+				},
+
+				{
+					name: "DeleteWhenDNE",
+					test: func(t *testing.T, d depot.Depot) {
+						name := "bob"
```

Comment:
super nit: use constants here rather than variables, even at the same scope. It 100% doesn't matter, but seems nice.